BookService/Logic/LibraryCollection.cs

```diff
@@ -0,0 +1,318 @@
+﻿using Logic.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Logic
+{
+    /// <summary>
+    /// Class-realization of the book's collection
+    /// </summary>
+    public class LibraryCollection : ILibrary
+    {
+        /// <summary>
+        /// List with books
+        /// </summary>
+        private List<Book> books;
+
+        /// <summary>
+        /// List with authors
+        /// </summary>
+        private List<Author> authors;
+
+        /// <summary>
+        /// List with Genres
+        /// </summary>
+        private List<Genre> genres;
+
+        /// <summary>
+        /// List with Book-Author pairs
+        /// </summary>
+        private List<BookAuthor> bookAuthors;
+
+        /// <summary>
+        /// List with Book-Genre pairs
+        /// </summary>
+        private List<BookGenre> bookGenres;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LibraryCollection"/> class.
+        /// </summary>
+        public LibraryCollection(IDataProvider data)
+        {
+            authors = new List<Author>(data.GetAuthors());
+            books = new List<Book>(data.GetBooks());
+            genres = new List<Genre>(data.GetGenres());
+            bookAuthors = new List<BookAuthor>(data.GetBooksAuthors());
+            bookGenres = new List<BookGenre>(data.GetBooksGenres());
+        }
+
+        /// <summary>
+        /// Get book value by it's index
+        /// </summary>
+        /// <param name="id">Index of the selected book</param>
+        /// <returns>Book by selected index</returns>
+        /// <exception cref="IndexOutOfRangeException">Throw when index out of
+        /// list's count range</exception>
+        public Author GetAuthorByIndex(int id)
+        {
+            return authors[id];
```

Comment:
As I have noticed before. Id and Index is different things. You shouldn't use Id like Index. 
Use return authors.SingleOrDefault(q=>q.Id == id); or FirstOrDefault or Find (it depended on what is more applicable for your logic)  in this case you don't need to handle that exception in the controller. And can handle null result or whatever find will return if such id would not found.